hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/security/RouterDelegationTokenSecretManager.java

```diff
@@ -91,283 +117,173 @@ private boolean shouldIgnoreException(Exception e) {
    * @param newKey DelegationKey
    */
   @Override
-  public void storeNewMasterKey(DelegationKey newKey) {
+  protected void storeNewMasterKey(DelegationKey newKey) throws IOException {
     try {
       federationFacade.storeNewMasterKey(newKey);
-    } catch (Exception e) {
-      if (!shouldIgnoreException(e)) {
-        LOG.error("Error in storing master key with KeyID: {}.", newKey.getKeyId());
-        ExitUtil.terminate(1, e);
-      }
+    } catch (YarnException e) {
+      e.printStackTrace();
+      throw new IOException(e); // Wrap YarnException as an IOException to adhere to storeNewMasterKey contract
     }
   }
 
   /**
-   * The Router Supports Remove the master key.
-   * During this Process, Facade will call the specific StateStore to remove the MasterKey.
-   *
-   * @param delegationKey DelegationKey
+   * no-op as expiry of stored keys is upto the state store in a stateless secret manager
    */
   @Override
   public void removeStoredMasterKey(DelegationKey delegationKey) {
-    try {
-      federationFacade.removeStoredMasterKey(delegationKey);
-    } catch (Exception e) {
-      if (!shouldIgnoreException(e)) {
-        LOG.error("Error in removing master key with KeyID: {}.", delegationKey.getKeyId());
-        ExitUtil.terminate(1, e);
-      }
-    }
+
   }
 
   /**
-   * The Router Supports Store new Token.
-   *
-   * @param identifier RMDelegationToken
-   * @param renewDate renewDate
-   * @throws IOException IO exception occurred.
+   * no-op as we are storing entire token with info in storeToken()
    */
   @Override
-  public void storeNewToken(RMDelegationTokenIdentifier identifier,
-      long renewDate) throws IOException {
-    try {
-      federationFacade.storeNewToken(identifier, renewDate);
```

Comment:
In the SQLFederationStateStore I see we are making 2 queries, one to add the new token, 2nd again to get the token. Is the 2nd query needed? (Not able to add review comment there so added here)

Reply:
yea thats not required. was planning to raise a seperate PR for changes in SQLFederationStateStore - will keep this PR focused on changes required for stateless delegation token management